leapp/utils/lock.py

```diff
@@ -0,0 +1,96 @@
+import os
+import fcntl
+import logging
+
+from leapp.config import get_config
+from leapp.exceptions import ProcessLockError
+
+
+def leapp_lock(lockfile=None):
+    return ProcessLock(lockfile=lockfile)
+
+
+def _acquire_lock(fd):
+    try:
+        fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        return True
+    except OSError:
+        return False
+
+
+def _read_pid(fd):
+    return os.read(fd, 20)
+
+
+def _write_pid(fd, pid):
+    os.write(fd, str(pid).encode('utf-8'))
```

Comment:
`_write_pid()` will corrupt the file if there is already data in it and `_clear_lock()` isn't called first so we should enforce doing the right thing by calling `_clear_lock()` here,  inside of `_write_pid()`.  
```suggestion
    _clear_lock(fd)
    os.write(fd, str(pid).encode('utf-8'))
```